redis/_compat.py

```diff
@@ -15,8 +20,12 @@ def select(rlist, wlist, xlist, timeout):
         while True:
             try:
                 return _select(rlist, wlist, xlist, timeout)
-            except InterruptedError:
-                continue
+            except InterruptedError as e:
+                # Python 2 does not define InterruptedError, instead
+                # try to catch an OSError with errno == EINTR == 4.
+                if hasattr(e, 'errno') and (e.errno == 4 or e.errno is None):
```

Comment:
Might be able to remove the `errno is None` check. I don't know if the `InterruptedError` exception sets `errno` on Python 3 but it might, we can try it! And I thought about doing the import but I'm not sure if all platforms define `errno.EINTR`. I know Windows has a lot of flakiness in the `errno` module. I'll use a `getattr` to be safe.